opts/opts_test.go

```diff
@@ -85,8 +90,44 @@ func TestListOpts(t *testing.T) {
 		t.Error("o.Get(\"baz\") == true")
 	}
 	o.Delete("foo")
-	if o.String() != "[bar]" {
-		t.Errorf("%s != [bar]", o.String())
+	if o.String() != "[bar bar]" {
+		t.Errorf("%s != [bar bar]", o.String())
+	}
+	listOpts := o.GetAll()
+	if len(listOpts) != 2 || listOpts[0] != "bar" || listOpts[1] != "bar" {
+		t.Errorf("Expected [[bar bar]], got [%v]", listOpts)
+	}
+	mapListOpts := o.GetMap()
+	if len(mapListOpts) != 1 {
+		t.Errorf("Expected [map[bar:{}]], got [%v]", mapListOpts)
+	}
+
+}
+
+func TestListOptsWithValidator(t *testing.T) {
+	// Re-using logOptsvalidator (used by MapOpts)
+	o := NewListOpts(logOptsValidator)
+	o.Set("foo")
```

Comment:
eh, never mind - same net result and I guess the there's only one type of error that can be returned so no need to check.